### PR TITLE
Role Panel & rune fix

### PR DESCRIPTION
**Pull request by DeityLink**
image

- [x] Traitor panel re-enabled, renamed to Role Panel, can be displayed from a mob's Player Panel
- [x] Role Panel lets you add any valid role, or remove them at will
- [x] Role Panel lets you add a role to a given faction. If said faction doesn't exist yet, it gets automatically created. Likewise, a role can be removed from a faction.
- [x] Role Panel lets you edit a Role or a Faction's objectives: adding new ones, editing and removing existing ones, or allowing admin to force their success.
- [x] Check Antagonist panel reworked to list factions (their objectives (editable) and their roles), then orphaned roles.
- [x] Fixed Runes not being recognized despite having proper word combinations.

TODO for a later PR
* write a custom select_target() for every /datum/objective/target
* write a safety proc that will make sure that you'll never have to protect/assassinate the same individual, or have to both survive and die.
* enthralling/apprenticeship automatically generates a custom faction whose leader is the vampire/wizard that created it.

### CODE REVIEW

File: code/datums/gamemode/role/role.dm [hunk 1]
```diff
@@ -279,11 +282,33 @@
 
 	to_chat(world, text)
```
Comment (ShiftyRail) [inline code/datums/gamemode/role/role.dm:286]: Probably not meant to stay there

File: code/datums/gamemode/objectives/objective.dm [hunk 2]
```diff
@@ -30,15 +34,17 @@
...
+				[admin_edit ? " - <a href='?src=\ref[src];obj_edit=\ref[O]'>(edit)</a> - <a href='?src=\ref[src];obj_delete=\ref[O]'>(remove)</a> - <a href='?src=\ref[src];obj_completed=\ref[O]'>(toggle:[current_completion ? "<font color='green'>SUCCESS" : "<font color='red'>FAILURE" ]</font>)</a>" : ""]
+				<br>"}
 			if(check_success)
-				dat += {"<BR>[O.IsFulfilled() ? "Success" : "Failed"]"}
-
-	return dat
\ No newline at end of file
+				dat += {"<BR>[current_completion ? "Success" : "Failed"]"}
+	if(admin_edit)
+		dat += "<a href='?src=\ref[src];obj_add=1'>(add personnal objective)</a>"
```
Comment (DamianX) [inline code/datums/gamemode/objectives/objective.dm]: >personnal

Comment (ghost) [reply code/datums/gamemode/objectives/objective.dm]: pshh... nothing head of personnal kid

Comment (DeityLink) [reply code/datums/gamemode/objectives/objective.dm]: cut me some slack, we say "personnel" in french >__>

File: code/datums/gamemode/role/role.dm (hunk 1 above)
Comment (DeityLink) [reply code/datums/gamemode/role/role.dm:286]: I think that the entire proc will have to be reworked at some point.

**Comment by DeityLink** [comment]
image

File: code/datums/gamemode/objectives/objective.dm (hunk 2 above)
Comment (ghost) [reply code/datums/gamemode/objectives/objective.dm]: Ouais ce n'est pas un grand problème

File: code/datums/gamemode/role/role.dm [hunk 3]
```diff
@@ -340,11 +365,17 @@
 
 // USE THIS INSTEAD (global)
 /datum/role/proc/RoleTopic(href, href_list, var/datum/mind/M, var/admin_auth)
+
+	if(!check_rights(R_ADMIN))
```
Comment (MadmanMartian) [inline code/datums/gamemode/role/role.dm]: changeling uses roletopic for the changeling menu

**Comment by DeityLink** [comment]
image

**Comment by DeityLink** [comment]
image
image

I fixed the typo in memory already.

**Pull request merged**